ee/presencedetection/auth.m

```diff
@@ -0,0 +1,68 @@
+//go:build darwin
+// +build darwin
+
+// auth.m
+#import <LocalAuthentication/LocalAuthentication.h>
+#include "auth.h"
+
+struct AuthResult Authenticate(char const* reason) {
+    struct AuthResult authResult;
+    LAContext *myContext = [[LAContext alloc] init];
+    NSError *authError = nil;
+    dispatch_semaphore_t sema = dispatch_semaphore_create(0);
+    NSString *nsReason = [NSString stringWithUTF8String:reason];
+    __block bool success = false;
+    __block NSString *errorMessage = nil;
+    __block int errorCode = 0;
+
+    // Use LAPolicyDeviceOwnerAuthentication to allow biometrics and password fallback
+    if ([myContext canEvaluatePolicy:LAPolicyDeviceOwnerAuthentication error:&authError]) {
+        [myContext evaluatePolicy:LAPolicyDeviceOwnerAuthentication
+            localizedReason:nsReason
+            reply:^(BOOL policySuccess, NSError *error) {
+                if (policySuccess) {
+                    success = true; // Authentication successful
+                } else {
+                    success = false;
+                    errorCode = (int)[error code];
+                    errorMessage = [error localizedDescription];
+                    if (error.code == LAErrorUserFallback || error.code == LAErrorAuthenticationFailed) {
+                        // Prompting for password
+                        [myContext evaluatePolicy:LAPolicyDeviceOwnerAuthentication
+                            localizedReason:@"Please enter your password"
```

Comment:
I don't think it's worth doing in this PR -- but maybe we would want to have this text provided by K2 in the future in another header, so that if we ever do localization for text K2 can handle it instead of us?

Reply:
This value never actually appears if select "use password", but it's still required. Maybe this only appears if you don' t have bio / touch id enabled. Either way, I set it to use same reason as bio / touch id.